pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1502,7 +1502,7 @@ cdef class _Timedelta(timedelta):
     @classmethod
     def _from_value_and_reso(cls, int64_t value, NPY_DATETIMEUNIT reso):
         # exposing as classmethod for testing
-        return _timedelta_from_value_and_reso(value, reso)
+        return _timedelta_from_value_and_reso(cls, value, reso)
```

Comment:
@jbrockmendel Is it still useful to have the `_timedelta_from_value_and_reso` function? After my changes it now has exactly the same signature as `_from_value_and_reso`

Reply:
i _think_ its more performant as it is bc cython doesn't support cdef classmethods. havent done any measurements on this though